server/src/main/java/io/druid/segment/realtime/RealtimeManager.java

```diff
@@ -246,58 +247,44 @@ public void run()
         // Delay firehose connection to avoid claiming input resources while the plumber is starting up.
         firehose = initFirehose();
 
-        long nextFlush = new DateTime().plus(intermediatePersistPeriod).getMillis();
+        final CommitterMaker committerMaker = CommitterMakers.fromFirehose(firehose);
         while (firehose.hasMore()) {
-          InputRow inputRow = null;
+          final InputRow inputRow;
           try {
-            try {
-              inputRow = firehose.nextRow();
+            inputRow = firehose.nextRow();
 
-              if (inputRow == null) {
-                log.debug("thrown away null input row, considering unparseable");
-                metrics.incrementUnparseable();
-                continue;
-              }
-            }
-            catch (ParseException e) {
-              log.debug(e, "thrown away line due to exception, considering unparseable");
+            if (inputRow == null) {
+              log.debug("thrown away null input row, considering unparseable");
               metrics.incrementUnparseable();
               continue;
             }
-
-            boolean lateEvent = false;
-            boolean indexLimitExceeded = false;
-            try {
-              lateEvent = plumber.add(inputRow) == -1;
-            }
-            catch (IndexSizeExceededException e) {
-              log.info("Index limit exceeded: %s", e.getMessage());
-              indexLimitExceeded = true;
-            }
-            if (indexLimitExceeded || lateEvent) {
-              metrics.incrementThrownAway();
-              log.debug("Throwing away event[%s]", inputRow);
-
-              if (indexLimitExceeded || System.currentTimeMillis() > nextFlush) {
-                plumber.persist(firehose.commit());
-                nextFlush = new DateTime().plus(intermediatePersistPeriod).getMillis();
-              }
-
-              continue;
-            }
-            final Sink sink = plumber.getSink(inputRow.getTimestampFromEpoch());
-            if ((sink != null && !sink.canAppendRow()) || System.currentTimeMillis() > nextFlush) {
-              plumber.persist(firehose.commit());
-              nextFlush = new DateTime().plus(intermediatePersistPeriod).getMillis();
-            }
-            metrics.incrementProcessed();
           }
           catch (ParseException e) {
-            if (inputRow != null) {
-              log.error(e, "unparseable line: %s", inputRow);
-            }
+            log.debug(e, "thrown away line due to exception, considering unparseable");
```

Comment:
yeah, the diff is a bit weird. The intent is to not change behavior in any way anyone would care about.